moonraker/plugins/power.py

```diff
@@ -522,6 +527,100 @@ async def set_power(self, state):
             raise self.server.error(msg) from None
         self.state = "on" if state else "off"
 
+class Indigo(PowerDevice):
+    def __init__(self, config):
+        super().__init__(config)
+        self.server = config.get_server()
+        self.addr = config.get("address")
+        self.output_name = config.get("output_name", "")
+        self.user = config.get("user", "")
+        self.password = config.get("password", "")
+
+    async def _send_indigo_command(self, command):
+        http_client = CurlAsyncHTTPClient()
+        if command == "on":
+            try:
+                out_cmd = f"devices/{self.output_name}?isOn=true"
+                url = f"http://{self.addr}/{out_cmd}"
+                info_cmd = f"devices/{self.output_name}.json"
+                info_url = f"http://{self.addr}/{info_cmd}"
+                requests.put(url, auth=HTTPDigestAuth(self.user, self.password))
+                response = await http_client.fetch( \
+                    info_url, \
+                    auth_mode='digest', \
+                    auth_username=self.user, \
+                    auth_password=self.password)
+                data = json_decode(response.body)
+            except Exception:
+                msg = f"Error sending indigo command: {command}"
+                logging.exception(msg)
+                raise self.server.error(msg)
+        elif command == "off":
+            try:
+                out_cmd = f"devices/{self.output_name}?isOn=false"
+                url = f"http://{self.addr}/{out_cmd}"
+                info_cmd = f"devices/{self.output_name}.json"
+                info_url = f"http://{self.addr}/{info_cmd}"
+                requests.put(url, auth=HTTPDigestAuth(self.user, self.password))
```

Comment:
You'd want to avoid using a synchronous client like requests inside an event loop function (in `async def` function).
It would block the event loop for the duration of the request, which is especially a problem when there's a network error (default timeout is something like 20 minutes) and will make everything look like it broke.

You should be able to use the async client like you did just below this:
```py
await http_client.fetch(url, method='PUT', auth_mode='digest', auth_username=self.user, auth_password=self.password)
```
https://www.tornadoweb.org/en/stable/httpclient.html#tornado.httpclient.HTTPRequest

Reply:
Thanks for the feedback!
This was initially the path I pursued, but I ran in to issues. After receiving the PUT, Indigo responds with a 303 (which is normal/expected). Tornado tries to follow the 303 (also expected), but strangely does another PUT rather than a GET. This results in a 500 from Indigo, which causes an error and the plugin stops responding. 

I attempted to direct tornado to ignore redirects with `follow_redirects=0`, but then the 303 was passed back and treated as an error which also caused the plugin to stop responding. 
Digging down farther, this seems to be a libcurl issue.  Looking at tornado, they call this out here:
https://github.com/tornadoweb/tornado/blob/6cdf82e927d962290165ba7c4cccb3e974b541c3/tornado/simple_httpclient.py#L644

As I'm using the FluiddPi (which is a Raspbian Lite based image), it's relying on "Buster" provided libcurl, which is 7.64.0 ( https://packages.debian.org/buster/libcurl4 ). 

At this point I gave up and used `requests`, but I do see the problem you present. I'm totally open to a better/more correct way to do this, I just don't know what that looks like. 

And now, for some logs showing what I'm seeing. 
[ignore_redirects.txt](https://github.com/Arksine/moonraker/files/6085020/ignore_redirects.txt)
[follow_redirects.txt](https://github.com/Arksine/moonraker/files/6085021/follow_redirects.txt)

Reply:
Did you try wrapping it in a try-catch statement?
e.g.
```python
try:
    await http_client.fetch(
        url, method='PUT', auth_mode='digest',
        auth_username=self.user, auth_password=self.password,
        follow_redirects=0)
except httpclient.HTTPError:
    # We expect an HTTPError due to 303 redirect
    pass
```

Reply:
This was exactly the cluebat I needed to be hit with! Thank you so much!

Reply:
Yay! A pleasure :-)